bot/exts/utilities/issues.py

```diff
@@ -139,7 +139,7 @@ async def fetch_issues(
             log.trace(f"PR provided, querying GH pulls API for additional information: {pulls_url}")
             async with self.bot.http_session.get(pulls_url) as p:
                 pull_data = await p.json()
-                if pull_data["draft"]:
+                if "draft" in pull_data and pull_data["draft"]:
                     emoji = Emojis.pull_request_draft
                 elif pull_data["state"] == "open":
```

Comment:
From what I have collected, `draft` should *always* be returned in a valid response from the API. Therefore if `draft` was missing, it likely means the entire response was something unexpected (perhaps empty). So we could expect each one of these elifs to fail as well. It may be prudent to utilize a try-except here, or explicitly catch a missing key from any if-elif block. From what I can see, the current will just cause the error to occur on the first elif, since that key would probably be missing as well.

Reply:
API docs for reference https://docs.github.com/en/rest/reference/pulls

Reply:
@fisher60 What should the emoji be set to if the pull_data is invalid?

Reply:
This would be fixed as part of #968, although I believe #778 would fix this as I think it moves all requests to a helper method which checks for a 200 response code and whatnot.

Reply:
I still think a try-except would be a good idea in case the pull data is malformed.  #778 still leaves the possibility of a KeyError.

Reply:
If its malformed, we shouldn't be doing anything with the pull data. A solution would be to use a helper method to get the data, deal with possible ratelimits, and check for a 200 status code.

Most of the calls are deferred to a helper method with #778, but not all of them, unfortunately. 

Additionally, this particular call to the api affecting this bug will be removed as a result of #968, as the data is already there with the initial issues api request.

The solution, if we are to implement it in this pull request, is to check if [L140](https://github.com/python-discord/sir-lancebot/blob/0d87e9ae6e44937af5e3b64a7cbe329660f9fff9/bot/exts/utilities/issues.py#L140) does not have a 200 response code, and deal with that accordingly.

Reply:
If its malformed, we shouldn't be doing anything with the pull data. A solution would be to use a helper method to get the data, deal with possible ratelimits, and check for a 200 status code.

Most of the calls are deferred to a helper method with #778, but not all of them, unfortunately. 

Additionally, this particular call to the api affecting this bug will be removed as a result of #968, as the data is already there with the initial issues api request.

The solution, if we are to implement it in this pull request, is to check if [L140](https://github.com/python-discord/sir-lancebot/blob/0d87e9ae6e44937af5e3b64a7cbe329660f9fff9/bot/exts/utilities/issues.py#L140) does not have a 200 response code, and deal with that accordingly.